Switch default build JDK to Java 12.

anuraaga opened the pull request:
Fixes #2141

trustin commented:
Could you also update the `release` task in `/build.gradle` to require Java 12? Maybe should we make the entire build require Java 12 rather than just the release process?

anuraaga commented:
Thanks indeed. Changed the release task to require Java 12, and added a check for normal compile that verifies >=12 (don't think we have to require exactly 12 for normal building during development)

trustin commented on build.gradle:
```diff
@@ -45,6 +45,16 @@ allprojects {
+task checkJavaVersion {
+    doLast {
+        if (JavaVersion.current().majorVersion.toInteger() < 12) {
+            throw new GradleException("The java version used ${JavaVersion.current()} is too old to " +
```
- java -> Java
- How about wrapping `${JavaVersion.current()}` with parens?

minwoox commented on build.gradle:
```diff
@@ -146,7 +156,7 @@ allprojects {
 
 // Require to use JDK 11 when releasing.
```
nit: JDK 12 😄
